aten/src/ATen/native/cuda/UpSample.cuh

```diff
@@ -253,7 +253,7 @@ __device__ __forceinline__ static int nearest_neighbor_bw_compute_source_index(
     int dst_index,
     int output_size) {
   const int src_index =
-      static_cast<int>(ceilf(dst_index * scale));
+      min(static_cast<int>(ceilf(dst_index * scale)), output_size - 1);
```

Comment:
given that subsequent loop is `for (int y = src_y; y < src_y_up; y++)`, I'd rather have `output_size` here, and no `+1` at the call site. Maximum returned index will be `output_size`, and it properly will never be accessed.